File: 프로그래머스/2/42584. 주식가격/주식가격.py
from collections import deque

def solution(prices):
    
    queue = deque(prices)
    answer = []
    while queue:
        current_price = queue.popleft()
        time = 0
        for price in queue:
            time += 1
            if current_price > price:
                break
        answer.append(time)
    return answer

File: 프로그래머스/2/42584. 주식가격/주식가격.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;

vector<int> solution(vector<int> prices) {
    vector<int> answer(prices.size(), 0);

    for(int i=0; i<prices.size(); i++) {
        int time = 0;
        for(int j=i+1; j<prices.size(); j++) {
            time++;
            if(prices[j] < prices[i]) {
                break;
            }
        }
        answer[i] = time;
    }
    return answer;
}